server/src/main/java/org/elasticsearch/action/admin/indices/get/GetIndexRequest.java

```diff
@@ -98,6 +94,10 @@ public GetIndexRequest(TimeValue masterTimeout) {
         super(masterTimeout, IndicesOptions.strictExpandOpen());
     }
 
+    /**
+     * Even though this action is executed on the local node, we still need to be able to read an incoming request as it's used across
+     * clusters.
+     */
```

Comment:
This one is worth discussing. It looks like the `GetIndexAction` is used across clusters:
https://github.com/elastic/elasticsearch/blob/a59c182f9f7e9d1bf3d6eecbc0e44f24ff91d053/server/src/main/java/org/elasticsearch/action/admin/indices/get/GetIndexAction.java#L19
which means we'll need to support transport serialization for both `GetIndexRequest` and `GetIndexResponse`. That in turn meant I had to remove `final` `LocalClusterStateRequest#writeTo` (see further down this PR). I don't really see a way around this but any suggestions are welcome.

We could also add `final` back into `LocalClusterStateReuqest#writeTo` and add a method like `writeToRemoteCluster` to something along those lines, to make it more explicit that that method is required for cross-cluster requests.